aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -1029,6 +1029,59 @@ Tensor transpose_nested(const Tensor& self, int64_t dim0, int64_t dim1) {
       self, sizemat_transposed, stridemat_transposed, std::vector<int64_t>(self_ptr->get_storage_offsets()));
 }
 
+Tensor squeeze_nested(const Tensor& self, int64_t dim) {
+  auto self_ptr = get_nested_tensor_impl(self);
+  int64_t ndim = self_ptr->dim();
+  int64_t wrapped_dim = at::maybe_wrap_dim(dim, ndim);
+  TORCH_CHECK(wrapped_dim > 0,
+  "squeeze(): For nested tensors, squeezing dimension 0 is not supported at the moment");
+  const Tensor& sizemat = self_ptr->get_nested_size_tensor();
+  const Tensor& stridemat = self_ptr->get_nested_stride_tensor();
+  // if tensor.size(dim) != 1 torch.squeeze will return the result, we do the same here
+  auto size_dim = sizemat.select(1, wrapped_dim - 1);
+  if (!at::all(size_dim == 1).item<bool>()) {
+    // detach to avoid triggering throw_error_if_base_and_tensor_are_same
+    return self.detach();
+  }
+  auto column_indices = sizemat.new_empty(ndim - 2);
```

Comment:
Does this function work with ndim is 2?

Reply:
Hm, this case returns a tensor iiuc, so we just error out?

Reply:
It's ok to throw an exception for this PR, take note in a separate issue and follow up

Reply:
In general, do nested tensors support having scalars as the nested elements?